### DISCUSSION TITLE
Compose syntax for player spawn Discord notifications

### DISCUSSION THREAD

**Discussion by AdamLMark** [lloesche/valheim-server-docker#457] [Q&A] [answered]
Hello, and thanks for the great project!

My docker-compose stack will not start with multiple `variable is not set` errors.

```
environment:
      - DISCORD_WEBHOOK=https://discord.com/api/webhooks/.....
      - VALHEIM_LOG_FILTER_CONTAINS_Spawned=Got character ZDOID from
      - ON_VALHEIM_LOG_FILTER_CONTAINS_Spawned='{ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="$l has arrived!"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }'
```

I saw the compose [syntax](https://github.com/lloesche/valheim-server-docker#notify-on-discord) section in the readme but it's difficult to compare since they are different functions.

Thanks! :)

EDIT: Thanks @drpcken and @revenghost! I created a [pull request](https://github.com/lloesche/valheim-server-docker/pull/480) to add valheim.env.default

**Comment by drpcken** [comment, marked as answer]
I had the exact same problem and ended up having to move the spawn message into my `valheim.env` file.  It had to do with string interpolation and my lack of knowledge, but that isn't an issue if you move it to your `valheim.env` file.   Here's my full env file:

```
SERVER_NAME=...
WORLD_NAME=...
SERVER_PUBLIC=false
SERVER_PASS=""
SUPERVISOR_HTTP=true
TZ=...
VALHEIM_PLUS=true
DISCORD_WEBHOOK=https://discord.com/api/webhooks/....
VALHEIM_LOG_FILTER_CONTAINS_Spawned=Got character ZDOID from
ON_VALHEIM_LOG_FILTER_CONTAINS_Spawned={ read l; l=${l:46}; l=${l// :*/}; msg="[$(date +'%b %d %T')] Player $l connected."; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Hugin\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }
```

**Reply by PyroghostX** [reply]
regex is best in the .env file

VALHEIM_LOG_FILTER_REGEXP_Spawned=\bGot character ZDOID from \b\w*\b : [^0]\d*\b:2\b
ON_VALHEIM_LOG_FILTER_REGEXP_Spawned={ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="Player $l is online!"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }

VALHEIM_LOG_FILTER_REGEXP_Death=\bGot character ZDOID from \b\w*\b : 0:0
ON_VALHEIM_LOG_FILTER_REGEXP_Death={ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="$l just died!"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }

VALHEIM_LOG_FILTER_CONTAINS_RandomEvent=Random event
ON_VALHEIM_LOG_FILTER_CONTAINS_RandomEvent={ read l; l=${l//*event set: /}; l=${l// :*/}; msg="Random event: $l"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }

**Reply by naanlizard** [reply]
Can you post this in code format? This is garbled to hell by the markdown parsing

**Reply by PyroghostX** [reply]
`VALHEIM_LOG_FILTER_REGEXP_Death=\bGot character ZDOID from \b.*\b : 0:0
ON_VALHEIM_LOG_FILTER_REGEXP_Death={ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="$l just died!"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }

VALHEIM_LOG_FILTER_CONTAINS_RandomEvent=Random event
ON_VALHEIM_LOG_FILTER_CONTAINS_RandomEvent={ read l; l=${l//*Random event set: /}; l=${l// :*/}; msg="Random event: $l"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }

VALHEIM_LOG_FILTER_REGEXP_Spawned=\bGot character ZDOID from \b.*\b : [^0]\d*\b:\b(1|2|3|4|5|6|7|8|9|10)\D
ON_VALHEIM_LOG_FILTER_REGEXP_Spawned={ read l; l=${l//*ZDOID from /}; l=${l// :*/}; msg="$l is online!"; curl -sfSL -X POST -H "Content-Type: application/json" -d "{\"username\":\"Valheim\",\"content\":\"$msg\"}" "$DISCORD_WEBHOOK"; }
`
@naanlizard

**Reply by PyroghostX** [reply]
Here it is attached
[valheim.txt](https://github.com/lloesche/valheim-server-docker/files/8643727/valheim.txt)
